opensrp-chw-hf/src/main/java/org/smartregister/chw/hf/sync/intent/HfSyncTaskIntentService.java

```diff
@@ -2,17 +2,49 @@
 
 import android.content.Intent;
 
+import com.evernote.android.job.JobManager;
+import com.evernote.android.job.JobRequest;
+import com.evernote.android.job.util.support.PersistableBundleCompat;
+
+import org.smartregister.AllConstants;
 import org.smartregister.chw.core.job.CloseExpiredReferralsServiceJob;
 import org.smartregister.chw.core.job.SyncTaskWithClientEventsServiceJob;
 import org.smartregister.chw.hf.sync.helper.HfTaskServiceHelper;
 import org.smartregister.sync.intent.SyncTaskIntentService;
 
+import java.util.concurrent.TimeUnit;
+
+import timber.log.Timber;
+
 public class HfSyncTaskIntentService extends SyncTaskIntentService {
     @Override
     protected void onHandleIntent(Intent intent) {
         HfTaskServiceHelper taskServiceHelper = HfTaskServiceHelper.getInstance();
         taskServiceHelper.syncTasks();
-        SyncTaskWithClientEventsServiceJob.scheduleJobImmediately(SyncTaskWithClientEventsServiceJob.TAG);
+        scheduleJob(SyncTaskWithClientEventsServiceJob.TAG, 15L, 5L);
```

Comment:
Why not schedule the task periodically like in https://github.com/opensrp/opensrp-client-chw-hf/blob/master/opensrp-chw-hf/src/main/java/org/smartregister/chw/hf/interactor/LoginInteractor.java#L42

Reply:
seems to have a bug in `scheduleJob()` method we are passing the `start` param as a minute and matching the condition with milliseconds. 

https://github.com/opensrp/opensrp-client-core/blob/d24dc795ad2e2583447ef59042c0afb0691ae154/opensrp-core/src/main/java/org/smartregister/job/BaseJob.java#L26
Line no 26 will always return false.

Reply:
Hi @owais-vd 
Yes, it appears the comparison `boolean toReschedule = start < TimeUnit.MINUTES.toMillis(15);` is evaluated as  `TimeUnit.MINUTES.toMinutes(15) < TimeUnit.MINUTES.toMillis(start)`. 
Won't this always result to  `toReschedule` as **`true`**?

Reply:
`TimeUnit.MINUTES.toMinutes(15)` always return minutes no conversion would be done because we are converting minutes to minutes.

For example: 
`TimeUnit.MINUTES.toMinutes(15)` return `15` and `TimeUnit.MINUTES.toMillis(15)` return `900000`. Therefore, `toReschedule` will always `true`.

Reply:
Yeah. The change makes sense